library/Zend/Http/Client.php

```diff
@@ -1192,6 +1192,13 @@ protected function _prepareHeaders()
             }
         }
 
+        // Set the request id header if the unique id environment variable
+        // is set.
+        $uniqueId = getenv('UNIQUE_ID');
+        if ($uniqueId && !isset($this->headers['x-request-id'])) {
+            $headers[] = "x-request-id: {$uniqueId}";
```

Comment:
On the previous line you are referring to $this->headers and then on the following line just $headers

Gotta sort that out